pkg/reconciler/ingress/ingress.go

```diff
@@ -224,28 +225,34 @@ func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ing *v1alpha1
 	}
 
 	// Now, remove the extra ones.
-	vses, err := r.virtualServiceLister.VirtualServices(ing.GetNamespace()).List(
-		labels.SelectorFromSet(labels.Set{networking.IngressLabelKey: ing.GetName()}))
-	if err != nil {
-		return fmt.Errorf("failed to get VirtualServices: %w", err)
+	selectors := map[string]string{
+		networking.IngressLabelKey: ing.GetName(), // VS created since 0.12
```

Comment:
// VS created between 0.12 and 0.13?

Reply:
No, it is from 0.12 on

Reply:
Why are we deleting them all then? Don't we only want to delete ones from < 0.14?